sskittenapi/catapi/catapi/models/cat_model.py

```diff
@@ -114,6 +119,21 @@ async def delete_cat(cat_id: dto.CatID) -> bool:
     return result.deleted_count == 1
 
 
+async def partial_update_cat(
+    partial_update_cat: dto.PartialUpdateCat, cat_id: dto.CatID
+) -> Optional[dto.Cat]:
```

Comment:
[question] Do you think that this function can return `None` or not?
And how do you think this function can be improved?

Reply:
I trying to refactor it with your hints.

Reply:
[nit] The return type of this function is not expected to be `None` because if no cat could be found, it will result in `CatNotFoundError`.
When returning the result of `find_one` directly, mypy will probably complain because the result of `find_one` **can be** `None`.
To deal with this, we can write the following:
```python
updated_cat = await find_one(cat_filter=dto.CatFilter(cat_id=cat_id))
assert updated_cat is not None
return updated_cat
```
This will tell mypy that we are never expecting the `updated_cat` to be `None` and thus the return type is not optional anymore.

Reply:
another option would be to keep this function to return `Optional[dto.Cat]`
and then check in domain whether the result of this function is `None` or not, and if `None` raise the exception you put below in the domain layer instead (this would be my recommendation as this way, you do not need to perform an `assert` but also you can make the domain layer keep the business logic, whereas modal layer should be a bit more concerned with DB related things) 
WDYT?

Reply:
https://github.com/SShakeeb/Kittenapi/pull/4